### ISSUE TITLE
Exclusive Modern Mode Feature

### ISSUE THREAD

**Issue by AndrewBogdanovTSS** [nuxt/nuxt#4552]
### What problem does this feature solve?

As a dev who don't need to support old browsers I would like to have a way to build *only* the modern version of the build so I could spend less time on generating the final assets

### What does the proposed changes look like?

I think in addition to `--modern` flag we can have smth. like ` --no-legacy` that will disable standard build generation

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8308">#c8308</a>)</em></sub></div>

**Comment by leekichko** [comment]
We really really really would like to see this, build times are terrible.

**Comment by ustun** [comment]
This would make things much faster in dev. I think most people should build against latest browsers, and the compile to older browsers at CI. Developers are paying a terrible price during development to support production. Of course, some sort of testing should be done before going live.

**Comment by koresar** [comment]
Serving static website (after `nuxt generate`). I can see that both legacy and modern bundles are loaded by the browser. What's the point of "modern" build if a user browser have to load legacy too?

**Comment by AndrewBogdanovTSS** [comment]
@koresar, are you sure that they both are actually loaded, maybe they only referenced twice?

**Comment by koresar** [comment]
Hello @Atinux

Sorry poking you. Please take a look at the screenshot above. Is it a bug?

TL;DR: the generated SPA-mode website downloads both legacy and modern bundles. Thus website gets twice larger.

**Comment by manniL** [comment]
@koresar Please file a new bug report ☺️ (at https://bug.nuxtjs.org/)

Please try it with `nuxt-edge` first (`yarn remove nuxt && yarn add nuxt-edge`) to see if the bug already has been fixed, eg via https://github.com/nuxt/nuxt.js/commit/ecf76d91f1bec8dcab8f2c7715e0e07a19d0b6fe

**Comment by AndrewBogdanovTSS** [comment]
@koresar did you run your code with explicit --modern flag? As far as I remember this was required until recently. Please, try to recreate it with Nuxt 2.4.2, I think it's already fixed

**Comment by koresar** [comment]
@AndrewBogdanovTSS looks fixed. Thank oyu.

**Comment by cypressious** [comment]
Any updates on this? Anthing that improves build times would be a welcome addition.

**Comment by manniL** [comment]
you could use `modern` only in prod for now @cypressious

**Comment by AndrewBogdanovTSS** [comment]
@manniL doing exactly that for now :)

**Comment by husayt** [comment]
Any ideas when was this fixed @koresar @manniL ? I see this still happening in 2.4.3 with generate:

>The resource ....js. was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.

I see borwser loading both sets of js files

**Comment by clarkdo** [comment]
@husayt Could you provide a repo ?

**Comment by clarkdo** [comment]
For anyone wants include modern bundles only, actually you don't need to use `build.modern`.

You can leverage `@nuxt/babel-preset-app` directly, increase babel transpiling level by：

```js
// nuxt.config
  build: {
    babel: {
      presets: ({ isServer }) => {
        return [[
          '@nuxt/babel-preset-app',
          {
            targets: isServer ? { node: 'current' } : {},
            modern: true
          }
        ]]
      }
    }
  }
```

**Comment by AndrewBogdanovTSS** [comment]
thanks for a workaround @clarkdo but it would be great to see a cleaner built in option that can be applied in nuxt.config

**Comment by clarkdo** [comment]
@AndrewBogdanovTSS Working on it recently 😸

**Comment by AndrewBogdanovTSS** [comment]
@clarkdo I tried your snippet but it didn't change build behavior in any way, still got 2 client builds happening instead of just modern one

**Comment by clarkdo** [comment]
With this config, you don't need to specify `modern`, because default client build will generate modern bundles by babel.

**Comment by AndrewBogdanovTSS** [comment]
@clarkdo so the actual "green" line means modern build now?

**Comment by husayt** [comment]
@clarkdo Here is the demo:

https://codesandbox.io/s/r4v8mr729p

I just added this to nuxt-config:
```
  mode: "spa",
  modern: true,
```

You can see browser loads both sets of js files modern and classic. I also get a warning in chrome:
>The resource https://r4v8mr729p.sse.codesandbox.io/_nuxt/app.js was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.
....

**Comment by clarkdo** [comment]
Yes, it's just a way to make bundles have a higher transpilation level, then bundles will be built with only supporting modern browsers.

**Comment by clarkdo** [comment]
@husayt Please have a look at my comment above, please remove `modern` in nuxt.config.

**Comment by husayt** [comment]
I tired various browsers. This happens in Chrome 69. Edge. IE11. Only Firefox 60 behaved as expected and did only load modern bundle.

**Comment by AndrewBogdanovTSS** [comment]
@clarkdo will it build all 3 bundles for production with such setup? I want `only-modern` only in dev mode

**Comment by clarkdo** [comment]
@AndrewBogdanovTSS We're discussing this, if the proposal gets agreed, I'll work on it asap

**Comment by clarkdo** [comment]
@husayt Your issue will be fixed in #5043

**Comment by AndrewBogdanovTSS** [comment]
@clarkdo any updates on this issue?

**Comment by danielroe** [comment]
This is now the default behaviour in Nuxt 3.
